pkg/jobs/jobfrontier/frontier.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package jobfrontier
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/isql"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/cockroachdb/cockroach/pkg/util/span"
+	"github.com/cockroachdb/errors"
+)
+
+// ShardFunc is a function that determines which shard a span belongs to.
+// It returns the shard key suffix to append to the base info key.
+// If sharding is not needed, this can be nil.
+type ShardFunc func(span roachpb.Span) string
```

Comment:
Concretely, what do you expect implementations of the ShardFunc to look like? Is this going to mimic the sharding of data into processors?

Reply:
removed